src/react/workflow/__tests__/ReplicationForm.test.tsx

```diff
@@ -0,0 +1,127 @@
+      "bootstrapList": [],
+      "bucketName": "replication-for-chapter-ux",
+      "endpoint": "http://s3.workloadplane.scality.local",
+      "region": "us-east-1",
+      "secretKey": "ICzSVrjcUJYYMiGU2TJV4hcCyuO0Ds6OJsh7D/Nyp/ua9zmCp2IxhBf38nv4N4x/9A6oZG11yiPkcFq7sYNWnepIuX+hJLlLN0RI/0MFv7WBhvA0Z5GN5zw24BtTiR6STgCxqaJ0kbE/2mc47TReap9PqiZ/vQZc4kSbBH+75EDTFqZsgKmEVGKNgKb9Llt56Ml4htdR3NJZ/Pd+BwiKMf1A6L9aroylkx8plarOkmM+9FS72lV2nDa/OStezRsNdsTDEMpXfApTewSBEE/Rq+7lgva8xrXZWz/V7f4L953m9i/lSd8ZhmCH2vpqowg+qGgVkVWMiSoAt5UpkzZBTg=="
+    },
+    "locationType": "location-scality-artesca-s3-v1",
+    "name": "chapter-ux",
+    "objectId": "4ab68d3f-9eec-11ec-ae58-6e38b828d159"
+  },
+  "us-east-1": {
+    "isBuiltin": true,
+    "locationType": "location-file-v1",
+    "name": "us-east-1",
+    "objectId": "95dbedf5-9888-11ec-8565-1ac2af7d1e53"
+  }
+}
+
+const server = setupServer(rest.post(
+  `${TEST_API_BASE_URL}/api/v1/instance/${instanceId}/accounts/${accountName}/workflows/${replicationId}`,
+  (req, res, ctx) => res(ctx.json([])),
+),);
+
+jest.mock("react-hook-form", () => ({
+  ...jest.requireActual("react-hook-form"),
+  useFormContext: () => ( {
+    register: jest.fn(),
+    getValues: jest.fn(),
+     formState: {
+       errors: {}
+     },
+  }),
+}));
+
+beforeAll(() => {
+  server.listen({ onUnhandledRequest: 'error' });
+  mockOffsetSize(200, 800);
+  jest.setTimeout(10_000);
+});
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+const WithFormProvider = ({children}) => {
+  const formMethods = useForm();
+  return <FormProvider {...formMethods}>{children}</FormProvider>
+}
+
+describe('ReplicationForm', () => {
+  it('should render a form for replication workflow', async () => {
+    try {
+
+      reduxRender(<WithFormProvider><ReplicationForm bucketList={S3BucketList} locations={locations}/></WithFormProvider>, {
+        networkActivity: {
+          counter: 0,
+          messages: List.of(),
+        },
+        instances: {
+          selectedId: instanceId,
+        },
+        auth: {
+          config: { features: [] },
+          selectedAccount: { id: accountId },
+        },
+        configuration: {
+          latest: {
+            locations,
+            endpoints: [],
+          },
+        },
+      });
+
+      await waitFor(() => screen.getByText(/General/i));
+
+      expect(screen.getByText('State')).toBeInTheDocument();
+      expect(screen.getByText('Source')).toBeInTheDocument();
+      expect(screen.getByText('Bucket Name')).toBeInTheDocument();
+      expect(screen.getByText('Filter (optional)')).toBeInTheDocument();
+      expect(screen.getByText('Prefix')).toBeInTheDocument();
+      expect(screen.getByText('Destination')).toBeInTheDocument();
+      expect(screen.getByText('Location Name')).toBeInTheDocument();
+
+    } catch (e) {
+      console.log('should render a form for replication workflow: ', e);
+      throw e;
+    }
+  });
+});
```

Comment:
Same as above we want to interract with the form and check if it is valid when all required inputs are correctly filled